### PR TITLE
Dot s block signatures

### PR DESCRIPTION
**Pull request by jgravelle-google**
This implements the s2wasm side of https://reviews.llvm.org/D25144 , which adds signatures to blocks in the .s format

### CODE REVIEW

**Comment by kripken** [comment]
lgtm

**Comment by dschuff** [comment]
OK I rebased this on top of #750 and updated the tests.

File: test/llvm_autogenerated/cfg-stackify.wast.orig [hunk 1]
```diff
@@ -0,0 +1,1128 @@
...
+      (call $something)
+      (br $label$0)
+    )
+  )
+  (func $test1 (param $0 i32)
+    (local $1 i32)
+    (set_local $1
+      (i32.const 1)
+    )
+<<<<<<< HEAD
```
Comment (jgravelle-google) [inline test/llvm_autogenerated/cfg-stackify.wast.orig]: Merge conflicted here, and in 15 more places

Looks to be only in the autogenerated tests though, so probably worth just regenerating them now that this is rebased

Comment (jgravelle-google) [inline test/llvm_autogenerated/cfg-stackify.wast.orig]: Oh right, the files are all-green and end with `.orig`

Lgtm

**Pull request merged**